src/envs/repeated_nextto_painting.py

```diff
@@ -0,0 +1,182 @@
+            _policy=self._Move_policy,
+            _initiable=utils.always_initiable,
+            _terminal=utils.onestep_terminal)
+
+    def simulate(self, state: State, action: Action) -> State:
+        assert self.action_space.contains(action.arr)
+        arr = action.arr
+        # Infer which transition function to follow
+        wash_affinity = 0 if arr[5] > 0.5 else abs(arr[5] - 0.5)
+        dry_affinity = 0 if arr[6] > 0.5 else abs(arr[6] - 0.5)
+        paint_affinity = min(abs(arr[7] - state.get(self._box, "color")),
+                             abs(arr[7] - state.get(self._shelf, "color")))
+        # NOTE: added move_affinity
+        move_affinity = sum(abs(val) for val in arr[2:])
+        affinities = [
+            (abs(1 - arr[4]), self._transition_nextto_pick_or_openlid),
+            (wash_affinity, self._transition_wash),
+            (dry_affinity, self._transition_dry),
+            (paint_affinity, self._transition_paint),
+            (abs(-1 - arr[4]), self._transition_nextto_place),
+            (move_affinity, self._transition_move),
+        ]
+        _, transition_fn = min(affinities, key=lambda item: item[0])
+        return transition_fn(state, action)
+
+    def _transition_nextto_pick_or_openlid(self, state: State,
+                                           action: Action) -> State:
+        x, y, z, _ = action.arr[:4]
+        next_state = self._transition_pick_or_openlid(state, action)
+        target_obj = self._get_object_at_xyz(state, x, y, z)
+        # if PaintingEnv pick was sucessful check if the object was next to
+        if target_obj is not None and state.get(
+                target_obj, "held") == 0.0 and next_state.get(
+                    target_obj, "held") == 1.0:
+            # Added cannot pick if obj is too far
+            if abs(state.get(self._robot, "y") \
+                - state.get(target_obj, "pose_y")) \
+                >= self.nextto_thresh:
+                return state
+        return next_state
+
+    def _transition_nextto_place(self, state: State, action: Action) -> State:
+        # Action args are target pose for held obj
+        _, y, _ = action.arr[:3]
+        next_state = self._transition_place(state, action)
+        #  Added restriction on place if not close enough to location
+        if abs(state.get(self._robot, "y") - y) \
+            >= self.nextto_thresh:
+            return state
+        return next_state
+
+    def _transition_move(self, state: State, action: Action) -> State:
+        # Action args are target y for robot
+        y = action.arr[1]
+        next_state = state.copy()
+        # Execute move
+        next_state.set(self._robot, "y", y)
+        held_obj = self._get_held_object(state)
+        if held_obj is not None:
+            next_state.set(held_obj, "pose_y", y)
+        return next_state
+
+    @property
+    def predicates(self) -> Set[Predicate]:
+        return {
```

Comment:
can we call the parent method to make obvious what new predicates are being added? so:
```return super().predicates | {self._NextTo, self._NextToBox, self._NextToShelf, self._NextToNothing}```